test/src/actions/CreateSpecialProblemAction.C

```diff
@@ -0,0 +1,53 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CreateSpecialProblemAction.h"
+
+#include "MooseApp.h"
+#include "FEProblemBase.h"
+
+registerMooseAction("MooseTestApp", CreateSpecialProblemAction, "create_problem_custom");
+
+template <>
+InputParameters
+validParams<CreateSpecialProblemAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addParam<std::string>("name", "Test Problem", "The name of the Problem object");
+  return params;
+}
+
+CreateSpecialProblemAction::CreateSpecialProblemAction(InputParameters parameters)
+  : Action(parameters)
+{
+}
+
+void
+CreateSpecialProblemAction::act()
+{
+  if (_problem)
+  {
+    // problem has been created by CreateProblemAction
+    if (_problem->type() != "MooseTestProblem")
```

Comment:
Hmm, this looks right to me.

Reply:
Yes, this particular action answers your question before. I think we want the input syntax to win always, right?

Reply:
This also means that the custom action needs to follow the similar pattern here for creating problems.